### ISSUE TITLE
Failsafe from rollbar-gem. OpenSSL::SSL::SSLError: \"SSL_connect returned=1 errno=0 state=error: certificate verify failed (unable to get local issuer certificate)

### ISSUE THREAD

**Issue by brightredchilli** [rollbar/rollbar-gem#980]
Hi, we are getting errors like these in production, and wondering if this would be the best place to report it.

```
[Rollbar] Error processing the item: OpenSSL::SSL::SSLError, SSL_connect returned=1 errno=0 state=error: certificate verify failed (unable to get local issuer certificate). 
Item: {"access_token"=>"***", "data"=>{"level"=>"error", "environment"=>"production", 
"body"=>{"message"=>{"body"=>"Failsafe from rollbar-gem. OpenSSL::SSL::SSLError: \"SSL_connect returned=1 errno=0 state=error: certificate verify failed (unable to get local issuer certificate)\" 
in /usr/local/lib/ruby/2.5.0/net/protocol.rb:44:in `connect_nonblock': error in process_item"}}, 
"notifier"=>{"name"=>"rollbar-gem", "version"=>"2.18.2"},
 "custom"=>{"orig_uuid"=>nil, "orig_host"=>nil}, "internal"=>true, "failsafe"=>true}}
```

A more detailed stacktrace:

```
              /usr/local/lib/ruby/2.6.0/net/
protocol.rb:  44:in `connect_nonblock'
              /usr/local/lib/ruby/2.6.0/net/
protocol.rb:  44:in `ssl_socket_connect'
                  /usr/local/lib/ruby/2.6.0/net/
http.rb: 996:in `connect'
                  /usr/local/lib/ruby/2.6.0/net/
http.rb: 930:in `do_start'
                  /usr/local/lib/ruby/2.6.0/net/
http.rb: 919:in `start'
                  /usr/local/lib/ruby/2.6.0/net/
http.rb:1470:in `request'
…ylight-core-4.1.2/lib/skylight/core/probes/
net_http.rb:  27:in `request'
…ms/elastic-apm-3.7.0/lib/elastic_apm/spies/
net_http.rb:  89:in `block in request'
…e/ruby/2.6.0/gems/elastic-apm-3.7.0/lib/
elastic_apm.rb: 290:in `with_span'
…ms/elastic-apm-3.7.0/lib/elastic_apm/spies/
net_http.rb:  79:in `request'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb: 555:in `block in do_post'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb: 616:in `handle_net_retries'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb: 555:in `do_post'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb: 533:in `send_item'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb: 204:in `process_item'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb: 244:in `block in process_from_async_handler'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb:  98:in `silenced'
               /usr/local/lib/ruby/2.6.0/forwardable.rb: 230:in `silenced'
…ruby/2.6.0/gems/rollbar-2.19.3/lib/rollbar/notifier.rb: 242:in `process_from_async_handler'
               /usr/local/lib/ruby/2.6.0/forwardable.rb: 230:in `process_from_async_handler'
…2.6.0/gems/rollbar-2.19.3/lib/rollbar/delay/sidekiq.rb:  19:in `perform'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 193:in `execute_job'
<truncated 17 additional frames>
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 160:in `block in process'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 133:in `block (6 levels) in dispatch'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/job_retry.rb: 110:in `local'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 132:in `block (5 levels) in dispatch'
…dle/ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/rails.rb:  43:in `block in call'
…esupport-5.2.3/lib/active_support/execution_wrapper.rb:  87:in `wrap'
…ems/activesupport-5.2.3/lib/active_support/reloader.rb:  73:in `block in wrap'
…esupport-5.2.3/lib/active_support/execution_wrapper.rb:  87:in `wrap'
…ems/activesupport-5.2.3/lib/active_support/reloader.rb:  72:in `wrap'
…dle/ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/rails.rb:  42:in `call'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 128:in `block (4 levels) in dispatch'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 254:in `stats'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 123:in `block (3 levels) in dispatch'
…uby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/job_logger.rb:  13:in `call'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 122:in `block (2 levels) in dispatch'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/job_retry.rb:  78:in `global'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 121:in `block in dispatch'
…le/ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/logger.rb:  10:in `with'
…uby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/job_logger.rb:  33:in `prepare'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 120:in `dispatch'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb: 159:in `process'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb:  78:in `process_one'
…ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/processor.rb:  68:in `run'
…ndle/ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/util.rb:  17:in `watchdog'
…ndle/ruby/2.6.0/gems/sidekiq-6.0.1/lib/sidekiq/util.rb:  26:in `block in safe_thread'
```

Because of the existence of "Failsafe from rollbar-gem", I assume that this issue is happening with the rollbar-gem, and just wanted to confirm.

Update: We haven't seen logs show up in the rollbar dashboard since these errors started.

**Comment by therabidbanana** [comment]
Also seeing this in our systems (running on Heroku's Heroku-18 stack, if that info is relevant). Based on error, guessing this is more likely an issue on the Rollbar API changing certificates than in the gem itself?

**Comment by KaraAJC** [comment]
Getting the same issue on one of my projects

**Comment by zaczheng** [comment]
+1 same issue

**Comment by d3mash** [comment]
Please use 👍  instead of spam-posting with +1s 😃

**Comment by KaraAJC** [comment]
getting this response when trying to call rollbar's API via curl:
```
curl https://api.rollbar.com
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
```
ran on both ruby 2.6.1 and 2.7.1

**Comment by wokenqingtian** [comment]
It just caused my delayed_jobs database to create unlimited tasks. 27000 records in a few minutes.

The rollbar version is 2.16.4

**Comment by waltjones** [comment]
Team at Rollbar is looking.

**Comment by derekyau** [comment]
This also caused our sidekiq queues to spike and create unlimited Rollbar delay jobs... we definitely need this fixed as soon as possible, thanks Rollbar.

Stack trace:
`Rollbar::Delay::Sidekiq	
{"access_token"=>"677f0098ab7349de896165adf22c29ba", "data"=>{"level"=>"error", "environment"=>"production", "body"=>{"message"=>{"body"=>"Failsafe from rollbar-gem. OpenSSL::SSL::SSLError: \"SSL_connect returned=1 errno=0 state=error: certificate verify failed (unable to get local issuer certificate)\" in /home/deployer/.rbenv/versions/2.6.5/lib/ruby/2.6.0/net/protocol.rb:44:in `connect_nonblock': error in process_item"}}, "notifier"=>{"name"=>"rollbar-gem", "version"=>"2.24.0"}, "custom"=>{"orig_uuid"=>nil, "orig_host"=>nil}, "internal"=>true, "failsafe"=>true}}`

**Comment by K1N5L4Y3R** [comment]
@derekyau I recommend you remove your `access_token` from your comment

**Comment by chiting** [comment]
@derekyau you can see still it from the `edited` history, I'd recommend to rotate it

**Comment by cabello** [comment]
I think it was fixed. Just tried a curl and it worked, errors seems to be reducing on our systems.

**Comment by derekyau** [comment]
@chiting @K1N5L4Y3R access_token was rotated, silly mistake thanks guys

**Comment by bmcniel** [comment]
From a sidekiq worker running heroku-18:

```
~ $ curl -vvv https://api.rollbar.com
* Rebuilt URL to: https://api.rollbar.com/
*   Trying 35.201.81.77...
* TCP_NODELAY set
* Connected to api.rollbar.com (35.201.81.77) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Unknown (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (OUT), TLS alert, Server hello (2):
* SSL certificate problem: unable to get local issuer certificate
* stopped the pause stream!
* Closing connection 0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
~ $ 
```

**Comment by K1N5L4Y3R** [comment]
FYI, I think this is the entry on Rollbar's status page for this issue: https://status.rollbar.com/incidents/39rm7hwmplpm

**Comment by nkipreos** [comment]
Hi guys, we are experiencing the same issue, any one here that has managed some kind of workaround for this?

**Comment by rubendinho** [comment]
This took down our redis instance since Sidekiq flooded it with retries and error reports (which then triggered more retries).  What are folks doing to temporarily Sidekiq reporting this without updating the Rollbar error reporting config?

**Comment by waltjones** [comment]
Should be working for everyone now.

**Comment by nkipreos** [comment]
And there goes our monthly sidekiq quota.

**Comment by waltjones** [comment]
Contact support and ask for your quota back. I think they'll be understanding.

**Comment by lmansur** [comment]
If your Sidekiq was flooded by the failsafe mechanism, you should try updating your gem. [2.23.0](https://github.com/rollbar/rollbar-gem/releases/tag/v2.23.0) supposedly fixes the issue, while [2.25.0](https://github.com/rollbar/rollbar-gem/releases/tag/v2.25.0) and [2.26.0](https://github.com/rollbar/rollbar-gem/releases/tag/v2.26.0) fixes other failsafe-related issues.

**Comment by brozek95** [comment]
Do you guys struggle with this issue from time to time? Rollbar gem version: 3.1.2
